doc/sg/c_cpp_coding_style.md

```diff
@@ -254,11 +271,43 @@ Furthermore, designated initializers do not play well with type deduction and ov
 As such, they are forbidden in the following contexts:
 - Do not call overloaded functions with a designated initializer: `overloaded({ .foo = 0 })`.
   Instead, disambiguate with syntax like `T var = { .foo = 0 }; overloaded(var);`.
-- Do not use designated initializers in any place where they would be used for type defuction.
+- Do not use designated initializers in any place where they would be used for type deduction.
   This includes `auto`, such as `auto var = { .foo = 0 };`, and a templated argument in a template function.
 
 It is recommended to only use designated initializers with types which use C-style declarations.
 
+### Naming
+**Structs, Classes and methods**
+As stated by the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Type_Names),the names of structs, classes and methods must be in `CamelCase` format.
+
+**Variable and class members naming**
+As stated by the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Variable_Names),the names of variables (including function parameters) and data members must be in `lower_snake_case` format. Data members of classes (but not structs) additionally have trailing underscores, unless the variable represents a constant that should follow the [rule](#function-enum-struct-and-typedef-naming).
+
+For example:
+```cpp
+// in animal.cpp
+public Class AnimalInfo{
+  ...
+private:
+  uint32_t number_of_paws_;
+  std::string name_;
+public:
+  static constexpr uint32_t kMaxNumOfPaws=100;
+
+  void SetAnimalName(std::string new_name);
+};
+```
+
+### Features
+
+Avoid using `C-style` features in C++ code, here are some examples:
+- Use `C++-style` casting (`static_cast`, `reinterpret_cast`, etc) rather than `C-style` casting.
+- Use the `const_expr` keyword to define constants rather than `const`.
+- Use the `nullptr` keyword rather than `NULL`.
+- Use `std::endl` rather than `'\n'`.
+- Use `new` and `delete` rather than `malloc()` and `free()`.
+- Use smart pointers rather then pointers. Refer to [Ownership and Smart Pointers](https://google.github.io/styleguide/cppguide.html#Ownership_and_Smart_Pointers) for more details.
```

Comment:
```suggestion
- Use smart pointers rather than pointers. Refer to [Ownership and Smart Pointers](https://google.github.io/styleguide/cppguide.html#Ownership_and_Smart_Pointers) for more details.
```

Reply:
Fixed